### ISSUE TITLE
EditorFunctionality change regarding ContentType/ResourceType

### ISSUE THREAD

**Issue by kkjamie** [dubit/unity-localisation#12]
The current type system is flawed.

**Context**
TLDR: ContentType is nothing more than a tag/label used for filtering at editor time.

The resource type is used primarily to filter. Each category is given a contentType enum, and then in the localised objects, each object class specifies which content it consumes using `[ResourceType]` attribute. This is used so that the custom inspector for these localised object component can show a dropdown of categories, but only the categories that have this `ContentType`.

The content type has no bearing on anything else, it doesn't change any other UI or change what's serialized. Everything is just a string, so audio/images are just paths to the resources they refer to.

Bearing this in mind the ContentType is nothing more than a tag used for filtering

**Problem**
The problem with the current system is that the content types are defined by an enum in the library meaning API users cannot extend this list without modifications to the library.

**Solution**
Given that the content type is just a tag/label filtering system (we shall refer to them as labels from this point onward)

My proposal is to allow categories to be assigned a number of labels. Then each component can specify optionally which labels they are interested in, and then the UI can only show categories that match those labels (or all categories if none are specified). The system would allow the user to add more labels to the system, but the system would have to come with some built in labels to support the built in localisation components.

The UI for adding labels can be primitve in the first iteration but ideally it would allow something like : https://bootstrap-tagsinput.github.io/bootstrap-tagsinput/examples/ where known labels are auto completed and new ones are added to the cache.

**Implementation**
Under the hood the labels would just be a list of strings. With fancier UI to show it.
On a static constructor in the unity editor we would compile a list of all known labels. this would be the built in labels as well as all labels we find in existing categories in the schema. This way we can provide auto complete.

In the  `AbstractLocalisedObject` class there can be an virtual `string[]` property that subclasses can override to define the labels it corresponds to. This would be wrapped in editor only scripting define.

**Comment by kkjamie** [comment]
ContentType/ResourceType has been removed. This is no longer relevant
